common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -246,62 +264,188 @@ func (s *localStoreCertProvider) getClientTLSSettings(isWorker bool) *config.Cli
 	} else if isWorker {
 		return s.legacyWorkerSettings // legacy config case when we use Frontend.Client settings
 	} else {
+		if s.tlsSettings == nil {
+			return nil
+		}
 		return &s.tlsSettings.Client // internode client case
 	}
 }
 
-func buildCAPoolFromData(caData []string) (*x509.CertPool, error) {
-	atLeastOneCert := false
-	caPool := x509.NewCertPool()
+func (s *localStoreCertProvider) Expiring(fromNow time.Duration,
+) (expiring CertExpirationMap, expired CertExpirationMap, errs []error) {
+
+	expiring = make(CertExpirationMap)
+	expired = make(CertExpirationMap)
+	errs = make([]error, 0)
+	when := time.Now().UTC().Add(fromNow)
```

Comment:
If you create some `TimeProvider` interface with `NowUTC() time.Time` func, and pass it as dependency to `localStoreCertProvider` you will be able to write bunch of unit test for this file,  easily moving time back and forth.

Reply:
Good idea. I will look at this as a follow-up change. Should help with testing cases of expired certs.